base/show.jl

```diff
@@ -1618,6 +1618,16 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int, quote_level::In
                 show_unquoted(io, arg1, indent, func_prec, quote_level)
             end
 
+        # ' with unicode suffix
+        elseif isa(func,Symbol) && length(func_args) == 1 && occursin(r"^'[\p{Lm}]+$", string(func))
```

Comment:
This should share an implementation with the case for the `'` head. You version is better since it handles operators (`(+)'`) and the existing code doesn't.